src/reconciler/hotReplacementRender.js

```diff
@@ -58,7 +58,7 @@ const haveTextSimilarity = (a, b) =>
 
 const equalClasses = (a, b) => {
   const prototypeA = a.prototype
-  const prototypeB = Object.getPrototypeOf(b.prototype)
+  const prototypeB = b.prototype // Object.getPrototypeOf(b.prototype)
```

Comment:
In React we are comparing unwrapped class with wrapped (real is behind getPrototypeOf). Should be controllable.